x/participationrewards/keeper/keeper.go

```diff
@@ -121,7 +121,7 @@ func (k Keeper) UpdateSelfConnectionData(ctx sdk.Context) error {
 	selfConnectionData, err := json.Marshal(types.ConnectionProtocolData{
 		ConnectionID: types.SelfConnection,
 		ChainID:      ctx.ChainID(),
-		LastEpoch:    ctx.BlockHeight() - 1,
+		LastEpoch:    ctx.BlockHeight() - 2, // reason why -2 works here.
```

Comment:
What does this comment mean?

Reply:
so this means to investigate the reason of off by 1.